agent/src/com/cloud/agent/Agent.java

```diff
@@ -274,6 +279,19 @@ public void start() {
             }
         }
         _shell.updateConnectedHost();
+
+        // In case of software based restart, GC to remove old instances
+        _executor.submit(new Runnable() {
```

Comment:
Can you explaining the goal of this code here?
Remove old instance of Java objects?

Reply:
Yes @rafaelweingartner, with the feature we've introduced a background thread that will perform software based restart of the agent. Look at the post renewal restart task. With this it will be easier to restart an agent without actually restarting the agent JVM process. This runnable is needed to GC old agent instance, we can remove this as well, but keeping it ensures that old agent is stopped+GC-ed.

Reply:
Got it. There is only one thing. We never now for sure that the GC has been executed. `System.gc` only suggests for the JVM that the GC can be executed, but there is no guarantees. Anyways, as long as the old objects are not used anymore we are safe to go.

Would you mind extracting this block of code to a method and document it with this explanation?

Reply:
Fixed.